src/nvim/regexp_defs.h

```diff
@@ -41,18 +43,38 @@
 #define NFA_ENGINE          2
 
 typedef struct regengine regengine_T;
+typedef struct regprog regprog_T;
+typedef struct reg_extmatch reg_extmatch_T;
```

Comment:
The whole definition of regmmatch_T is needed before the include of "buffer_defs.h". Should i still split it into a typedef struct and struct?